packages/myst-cli/src/process/site.ts

```diff
@@ -139,17 +138,19 @@ export async function loadProject(
 
 export function selectPageReferenceStates(session: ISession, pages: { file: string }[]) {
   const cache = castSession(session);
-  const pageReferenceStates: PageReferenceStates = pages
+  const pageReferenceStates: ReferenceState[] = pages
     .map((page) => {
-      const selectedFile = selectors.selectFileInfo(session.store.getState(), page.file);
-      return {
-        state: cache.$internalReferences[page.file],
-        file: page.file,
-        url: selectedFile?.url ?? null,
-        dataUrl: selectedFile?.dataUrl ?? null,
-      };
+      const state = cache.$internalReferences[page.file];
+      if (state) {
+        const selectedFile = selectors.selectFileInfo(session.store.getState(), page.file);
+        if (selectedFile?.url) state.url = selectedFile.url;
```

Comment:
(This is just consuming the change of these properties living directly on the `ReferenceState`, rather than a parent state that included them plus the state......)